### PR TITLE
Add: search conditions `OR` and `AND`

### PR DESCRIPTION
**Pull request by MimiDumpling**
Search now includes the conditions `OR` and `AND`. This PR is dependent on https://github.com/getsentry/sentry/pull/19718 AND replaces https://github.com/getsentry/sentry-docs/pull/1857 

image

### CODE REVIEW

**Comment by dashed** [comment]
This is dependent on this being merged https://github.com/getsentry/sentry/pull/19718

File: gatsby/src/docs/workflow/search/index.mdx [hunk 1]
```diff
@@ -32,6 +32,24 @@ The tokens `is:resolved` and `user.username:"Jane Doe"` are standard search toke
 
 The token `example error` is utilizing the optional raw search and is passed as part of the issue search query (which uses a CONTAINS match similar to SQL). When using the optional raw search, you can provide _one_ string, and the query uses that entire string.
 
+### Using `OR` and `AND`
+
+NOTE: `OR` and `AND` search conditions are only available for [Discover](/performance-monitoring/discover-queries/), [Performance](/performance-monitoring/performance/), and [Metric Alerts](/workflow/alerts-notifications/alerts/#metric-alerts).
+
+Use `OR` and `AND` between tokens, and use parentheses `()` to group conditions. `AND` can also be used between non-aggregates and aggregates. However, `OR` cannot.
```
Comment (adhiraj) [inline gatsby/src/docs/workflow/search/index.mdx:39]: @evanh are aggregates searches only applicable to discover?

Comment (dashed) [reply gatsby/src/docs/workflow/search/index.mdx:39]: @adhiraj They're applicable to Discover, Performance, and Performance Metric Alerts.

Comment (MimiDumpling) [reply gatsby/src/docs/workflow/search/index.mdx:39]: Adi and I were thinking of defining the difference between an aggregate and a non-aggregate. How's this:

Aggregate functions: groups data on numerical scales (for example, count():>100)
Non-aggregate functions: groups data based on specific attributes (for example, user.username:Jane)

Comment (dashed) [reply gatsby/src/docs/workflow/search/index.mdx:39]: I think that works.

Comment (MimiDumpling) [reply gatsby/src/docs/workflow/search/index.mdx:39]: @evanh do you think we can improve on the definitions of aggregate vs non-aggregate functions? I drafted some definitions above.

**Pull request merged**